src/PostgREST/Config.hs

```diff
@@ -62,9 +65,13 @@ import PostgREST.Types            (JSPath, JSPathExp (..),
                                    LogLevel (..))
 import Protolude                  hiding (concat, hPutStrLn,
                                    intercalate, null, replace, take,
-                                   toS, (<>))
+                                   toList, toLower, toS, (<>))
 import Protolude.Conv             (toS)
 
+-- | Command line interface options
+data CLI = CLI
+  { cliDumpConfig :: Bool
+  , cliPath       :: FilePath }
```

Comment:
As we are hoing to add options here that are mutually exclusive (e.g. dump schema cache), can we make this s sym type instead of bool?

Reply:
E.g. `data CliMode = Run | DumpConfig` (`| DumpSchemaCache` etc.)

Reply:
Very good point, Remo. I think `--dump-config` as an option is wrong to start with, when those things are mutually exclusive.

Two ideas to make this better:
- Use `--dump=config` and `--dump=schema` - this is mutually exclusive from the start. Then have a `CliDump = DumpConfig | DumpSchema | DumpNothing`.
- Use a "sub command" based approach. So have subcommands like `postgrest run`, `postgrest dump-config` and `postgrest dump-schema`. Of course the default without command would be `run`. Those commands would be mutually exclusive as well by design. Then have `CliCmd = CmdRun | CmdDumpConfig | CmdDumpSchema` like you suggested. This would be a bit more flexible in case we're adding other commands in the future. Not sure how easy the "default to run without command" is with the cli parser.

I think I like the `--dump=...` option better. WDYT?

Reply:
--dump-config if good I think! Most straight forward. Optparse applicative will be able to enforce a consistent command (see 'alternatives' where a sum type is the basis and 'flags' in https://github.com/pcapriotti/optparse-applicative)

Reply:
Ok, I went with a `--dump-config` flag and `data Command = CmdRun | CmdDumpConfig` for now.